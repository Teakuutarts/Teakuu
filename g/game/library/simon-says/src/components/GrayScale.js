import styled from 'styled-components';

const GrayScale = styled.div`
  transition: 0.7s;
  filter: grayscale(100%);
  ${props => !props.disabled ? '' : `
    filter: none;
  `}
`;

export default GrayScale;

