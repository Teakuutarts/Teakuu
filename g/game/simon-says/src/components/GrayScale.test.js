import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import GrayScale from './GrayScale';

it('renders correctly', () => {
  const wrapper = shallow(
    <GrayScale />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders correctly when disabled', () => {
  const wrapper = shallow(
    <GrayScale disabled={true} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

