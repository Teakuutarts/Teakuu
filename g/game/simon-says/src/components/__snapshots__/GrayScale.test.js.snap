exports[`test renders correctly 1`] = `
<div
  className="heKzYt" />
`;

exports[`test renders correctly when disabled 1`] = `
<div
  className="jmlCCS"
  disabled={true} />
`;
